package/gluon-statistics-mcast/src/respondd.c

```diff
@@ -0,0 +1,478 @@
+/*
+  Copyright (c) 2021, Linus Lüssing <linus.luessing@c0d3.blue>
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+
+    1. Redistributions of source code must retain the above copyright notice,
+       this list of conditions and the following disclaimer.
+    2. Redistributions in binary form must reproduce the above copyright notice,
+       this list of conditions and the following disclaimer in the documentation
+       and/or other materials provided with the distribution.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+  FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+  DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+  OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+
+#include <respondd.h>
+
+#include <json-c/json.h>
+#include <libgluonutil.h>
+
+#include <dirent.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/un.h>
+#include <unistd.h>
+
+#define BPFCOUNTD_SOCKS_DIR "/var/run/bpfcountd/gluon-sockets"
+
+enum capdir {
+	CAP_IN,
+	CAP_OUT,
+	__CAP_COUNT,
+};
+
+static const char *dirstr[__CAP_COUNT] = {
+	[CAP_IN] = "in",
+	[CAP_OUT] = "out",
+};
+
+static int check_socket_suffix(char *socketname)
```

Comment:
This should use `const char *`. There are lots more arguments that should be `const` in the code.

Reply:
Urgh, I just realized how evil strchr() is. Why the hell can it return a "char *" from a "const char *"??

For socketname it seems okay to change it to const (as long as I'm not missing something). But with parse_line() I probably shouldn't change "char *line" to "const char *line", as there is a `*start = '\0';` in there, which modifies `*line`.